segmentstore/contracts/src/main/java/io/pravega/segmentstore/contracts/SegmentApi.java

```diff
@@ -239,4 +240,14 @@
      * failed, the future will be failed with the causing exception.
      */
     CompletableFuture<Void> truncateStreamSegment(String streamSegmentName, long offset, Duration timeout);
+
+    /**
+     * Lists all the storage chunks for the given StreamSegment.
+     *
+     * @param streamSegmentName The name of the StreamSegment.
+     * @param bufferSize        The size of the buffer maintained when transferring data from storage.
+     * @param timeout           Timeout for the operation.
+     * @return
+     */
+    CompletableFuture<List<ExtendedChunkInfo>> listStorageChunks(String streamSegmentName, int bufferSize, Duration timeout);
```

Comment:
`getExtendedChunkInfo()` might be better name.
int bufferSize, is implementation detail , doesn't need to be in API.

Reply:
I've changed this name.